account/payments.go

```diff
@@ -945,7 +946,14 @@ func (a *Service) watchPayments() {
 				a.log.Criticalf("Failed to receive an invoice : %v", err)
 				return
 			}
-			if invoice.Settled {
+			if invoice.State == lnrpc.Invoice_SETTLED {
+				err := a.waitHtlcsSettled(ctx, invoice)
+				if err != nil {
+					// NOTE: On error, we'll notify invoice settled anyway.
+					// Not returning here.
+					a.log.Criticalf("Failed wait for htlc settlement: %v", err)
```

Comment:
This will shutdown (or even panic), perhaps we can only log Error and continue?

Reply:
Whoops